src/__internal__/form-field/form-field.spec.tsx

```diff
@@ -0,0 +1,247 @@
+      }
+    );
+
+    it("renders label with htmlFor prop that is the id of the input", () => {
+      const id = "foo";
+      const wrapper = render({
+        id,
+        name: "foo",
+        label: "Name",
+      });
+      expect(wrapper.find(`input[id='${id}']`).exists()).toBe(true);
+      expect(wrapper.find(`label[htmlFor='${id}']`).exists()).toBe(true);
+    });
+
+    it("passes the tooltipId to the Label id prop", () => {
+      const tooltipId = "test-help-id";
+      expect(
+        render({ tooltipId, label: "test label" }).find(Label).prop("tooltipId")
+      ).toBe(tooltipId);
+    });
+
+    describe("when validationRedesignOptIn is true", () => {
+      it("does not set the validation props on the Label", () => {
+        const { error, warning, info } = render({
+          label: "test label",
+          error: true,
+          warning: true,
+          info: true,
+          validationRedesignOptIn: true,
+        })
+          .find(Label)
+          .props();
+
+        expect(error).toEqual(false);
+        expect(warning).toEqual(false);
+        expect(info).toEqual(false);
+      });
+    });
+
+    describe("when adaptiveLabelBreakpoint prop is set", () => {
+      describe("when screen bigger than breakpoint", () => {
+        beforeEach(() => {
+          mockMatchMedia(true);
+        });
+
+        it("should pass labelInline to its children", () => {
+          const wrapper = render({
+            label: "Name",
+            labelInline: true,
+            fieldHelp: "Help",
+            adaptiveLabelBreakpoint: 1000,
+          });
+
+          expect(wrapper.find(FieldLineStyle).props().inline).toEqual(true);
+          expect(wrapper.find(Label).props().inline).toEqual(true);
+          expect(wrapper.find(FieldHelp).props().labelInline).toEqual(true);
+        });
+      });
+
+      describe("when screen smaller than breakpoint", () => {
+        beforeEach(() => {
+          mockMatchMedia(false);
+        });
+
+        it("should pass labelInline to its children", () => {
+          const wrapper = render({
+            label: "Name",
+            labelInline: true,
+            fieldHelp: "Help",
+            adaptiveLabelBreakpoint: 1000,
+          });
+
+          expect(wrapper.find(FieldLineStyle).props().inline).toEqual(false);
+          expect(wrapper.find(Label).props().inline).toEqual(false);
+          expect(wrapper.find(FieldHelp).props().labelInline).toEqual(false);
+        });
+      });
+    });
+  });
+
+  describe("with fieldHelp", () => {
+    const fieldHelpId = "mock-id";
+    const fieldHelp = "Help me!";
+
+    it("renders field help element with correct id", () => {
+      const wrapper = render({
+        fieldHelpId,
+        fieldHelp,
+        labelWidth: 20,
+      });
+      expect(wrapper.find(`[id='${fieldHelpId}']`).exists()).toBe(true);
+    });
+
+    it("when fieldHelpInline and labelInline props are true, field help is rendered inline with input", () => {
```

Comment:
This test isn't really testing that it's inline as the the display value of the fieldHelp isn't what would make it inline etc. I think it would be better to assert that when `fieldHelpInline` is set the `FieldHelp` is a child of the `FieldLineStyle` and when it's false it's not etc. This would work as you've already asserted that when `labelInline` is set it `FieldLineStyle` is flexed and so on

```
it("when fieldHelpInline and labelInline props are true, field help is rendered inline with input" (fieldHelpInline) => {
const wrapper = render({
        fieldHelpId,
        fieldHelp,
        labelWidth: 20,
        fieldHelpInline: true,
        labelInline: true,
      });

 assertStyleMatch({ display: "flex",}, wrapper.find(FieldLineStyle));
 expect(wrapper.find(FieldLineStyle).find(FieldHelp).exists()).toBe(true);
});

it("when fieldHelpInline is false and labelInline props is true, field help is not rendered inline with input" (fieldHelpInline) => {
const wrapper = render({
        fieldHelpId,
        fieldHelp,
        labelWidth: 20,
        fieldHelpInline: false,
        labelInline: true,
      });

 assertStyleMatch({ display: "flex",}, wrapper.find(FieldLineStyle));
 expect(wrapper.find(FieldLineStyle).find(FieldHelp).exists()).toBe(false);
 });
```